server/src/test/java/org/elasticsearch/cluster/routing/allocation/BalanceConfigurationTests.java

```diff
@@ -71,7 +68,7 @@ public void testIndexBalance() {
         settings.put(BalancedShardsAllocator.SHARD_BALANCE_FACTOR_SETTING.getKey(), replicaBalance);
         settings.put(BalancedShardsAllocator.THRESHOLD_SETTING.getKey(), balanceTreshold);
 
-        AllocationService strategy = createAllocationService(settings.build(), new NoopGatewayAllocator());
+        AllocationService strategy = createAllocationService(settings.build(), new TestGatewayAllocator());
```

Comment:
why were these changes needed?

Reply:
BalanceConfigurationTests used some oversimplified statement about cluster state: it could pick up UNASSIGNED shard and assign it (w/o allocate stale) as a new primary. That hit assertion i’ve added on checking allocation id.